hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/FiveToSixUpgradeHandler.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.table.upgrade;
+
+import org.apache.hadoop.fs.Path;
+import org.apache.hudi.common.config.ConfigProperty;
+import org.apache.hudi.common.engine.HoodieEngineContext;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.table.timeline.HoodieInstant;
+import org.apache.hudi.common.table.timeline.HoodieTimeline;
+import org.apache.hudi.config.HoodieWriteConfig;
+import org.apache.hudi.exception.HoodieUpgradeDowngradeException;
+import org.apache.hudi.table.HoodieTable;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Upgrade handle to assist in upgrading hoodie table from version 5 to 6.
+ */
+public class FiveToSixUpgradeHandler implements UpgradeHandler {
+
+  @Override
+  public Map<ConfigProperty, String> upgrade(HoodieWriteConfig config, HoodieEngineContext context, String instantTime, SupportsUpgradeDowngrade upgradeDowngradeHelper) {
+    HoodieTable table = upgradeDowngradeHelper.getTable(config, context);
+    HoodieTableMetaClient metaClient = table.getMetaClient();
+    // delete compaction requested file from .aux
+    HoodieTimeline compactionTimeline = metaClient.getActiveTimeline().filterPendingCompactionTimeline()
+        .filter(instant -> instant.getState() == HoodieInstant.State.REQUESTED);
+    compactionTimeline.getInstants().stream().forEach(instant -> {
+      String fileName = instant.getFileName();
+      try {
+        metaClient.getFs().delete(new Path(metaClient.getMetaAuxiliaryPath(), fileName));
```

Comment:
@Mulavar : We need to basically look at all caller of metaClient.getMetaAuxiliaryPath() and remove the usage of metaAuxiliaryPath since we should not no longer write to them.  Also in HoodieTableMetaClient.initTableAndGetMetaClient, can you remove the creation of .aux folder

Reply:
I found that some other feature like bootstrap index also use the .aux directory, refer to org.apache.hudi.common.table.HoodieTableMetaClient#BOOTSTRAP_INDEX_ROOT_FOLDER_PATH. Maybe we should remove the caller of metaClient.getMetaAuxiliaryPath() related to compaction logic?

Reply:
Yes, Good point. We should only cleanup compaction plans in .aux folder and remove compaction only callers